### PR TITLE
COMMERCE-6843 Migrate account-rest endpoints to headless-admin-user module

### PR DESCRIPTION
**Pull request by luca-pellizzon**

### CODE REVIEW

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
+                        The account's email domains. Users assigned to this account generally will have email addresses
+                        under one of these domains.
+                    items:
+                        type: string
+                    type: array
+                externalReferenceCode:
+                    # @review
+                    description:
+                        The optional external key of this account.
+                    type: string
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml:32]: @luca-pellizzon this needs to be `readOnly: true`

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@luca-pellizzon there is at least the one schema change needed (see my inline comment), but I would double-check any changes you find from this reflog and apply as needed with a regen: ```git log --grep LPS-135184 --grep LPS-135587 -- 'modules/apps/account/account-rest-*'```

**Review by drewbrokke — APPROVED** [review]

**Comment by drewbrokke** [comment]
ci:forward

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java [hunk 2]
```diff
@@ -420,9 +406,7 @@ protected Account testPutAccount_addAccount() throws Exception {
 	protected Account testPutAccountByExternalReferenceCode_addAccount()
 		throws Exception {
 
-		return accountResource.putAccountByExternalReferenceCode(
-			StringUtil.toLowerCase(RandomTestUtil.randomString()),
-			randomAccount());
+		return _postAccount();
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java:409]: @luca-pellizzon these are additional changes lost from Amos. Please double-check any changes found in this reflog: `git log --grep LPS-135184 --grep LPS-135587 -- 'modules/apps/account/account-rest-*'`.

**Review by drewbrokke — CHANGES_REQUESTED** [review]

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Pull request closed without merging**